### PR TITLE
Remove use of type_manager in apptools.naming, remove type_manager, and remove naming.adapter

### LINKED ISSUE #136: Remove apptools.type_manager?
The `apptools.type_manager` module has a lot of overlap with the Traits adaptation system since the latter gained the ability to have multiple adapter registries.

It is, however, used by `apptools.naming` which is in turn used by Mayavi, so we can't simply remove.

The correct way forward is probably to investigate whether `apptools.naming` can use Traits adaptation, and whether Mayavi uses the type manager's objects directly via the naming module

### PR DESCRIPTION
**Pull request by aaronayres35**
fixes #136 
part of #146 

In this PR, `apptools.naming` is converted to use `traits.adaptation`, and the `apptools.type_manager` subpackage is removed.  I did these in the same PR, but if a reviewer would prefer I can undo the last commit and defer the removal of `type_manager` to another PR.  
It is important to note that the code relating to `type_manager` in `apptools.naming` was largely untested, and from what I can tell, unused.  There is one example using it, but it seems to be broken for other reasons(? I need to investigate).

I also looked for other non-apptools uses of `type_manager` and could not find any.  Also, as the issue suggests I looked into Mayavi to see if it used the type manager's objects directly via the naming module, and it does not appear to do so. 

**Checklist**
- [x] Add a news fragment if this PR is news-worthy for end users. (see docs/releases/README.rst)

### CODE REVIEW

**Review by kitchoi — COMMENTED** [review]
I'm afraid I am not comfortable with the changes as the adapters logic are not covered in tests.

File: apptools/naming/context_adapter.py [hunk 1]
```diff
@@ -16,18 +16,12 @@
...
     """ The base class for all context adapters. """
-
-    #### 'ContextAdapter' interface ###########################################
-
-    # The object that we are adapting.
-    adaptee = Any
-
-    # The context that the object is in.
-    context = Instance(Context)
+    pass
```
[the remark this answers is not in the document]
Comment (kitchoi) [reply apptools/naming/context_adapter.py:27]: Another possibility is that it has never worked on master, then we can just kill them?

**Comment by kitchoi** [comment]
For future reference, these are the observations made that result in the type manager being removed:

The initial observation is that `apptools.naming.api.explore` is used by mayavi, here: https://github.com/enthought/mayavi/blob/9c128635943e28513776dec68492ffd0268c28d1/mayavi/plugins/mayavi_ui_plugin.py#L207-L213
This makes the `explore` function available from mayavi2's IPython prompt when the application is run with wxPython backend. Further observations:

- The functionality does not work on wxPython4 because pyface Tree does not work on wxPython4 as of Pyface 7.1.0 (see https://github.com/enthought/pyface/pull/810 which fixes it)
- The functionality does work on Python 2 and wxPython 3 (@aaronayres35 please confirm).

However, the code path of `apptools.naming.api.explore` never requires the `type_manager` adaptation in `apptools.naming.context`.

Details of the static code analysis
---------
The adaptation is called here: https://github.com/enthought/apptools/blob/8af7199ea4d8c9670d87d12d7d9e8651aa6889b8/apptools/naming/context.py#L755
Which is called here:
https://github.com/enthought/apptools/blob/8af7199ea4d8c9670d87d12d7d9e8651aa6889b8/apptools/naming/context.py#L706
Which is, conditionally, called by `Context.bind`, `rebind`, `unbind`, `lookup`, `lookup_binding`, `list_bindings` and `list_names`. 

In the `explorer` UI, `Context.lookup_binding`, `Context.list_bindings` and `Context.list_names` are called. 

For `lookup_binding`, this lives in a method [`ensure_visible`](https://github.com/enthought/apptools/blob/58fbef63179917419525fa175a2a8daa96ef33ac/apptools/naming/ui/naming_tree.py#L49) that is not called anywhere in apptools, and is also not called by the `NodeTree` base class (the method is not found anywhere in pyface source).  So it looks to be dead code.

For `list_bindings` and `list_names`, they are always called with an empty string, e.g.:
https://github.com/enthought/apptools/blob/8af7199ea4d8c9670d87d12d7d9e8651aa6889b8/apptools/naming/ui/context_node_type.py#L53

When the argument is an empty string, the `Context._get_next_context` requiring type managers is not called:
https://github.com/enthought/apptools/blob/8af7199ea4d8c9670d87d12d7d9e8651aa6889b8/apptools/naming/context.py#L516-L534

So the code path never requires type managers.

Further confirmation with manual testing
-----
(1) `explore` does not add any adapters prior to instantiating the `Explorer`:
https://github.com/enthought/apptools/blob/8af7199ea4d8c9670d87d12d7d9e8651aa6889b8/apptools/naming/ui/explorer.py#L26-L35
If the adapters are required, the type manager needs to be defined on the context, like in the example:
https://github.com/enthought/apptools/blob/8af7199ea4d8c9670d87d12d7d9e8651aa6889b8/examples/naming/explorer.py#L94-L95

(2) Even if we do define the adapters, they don't make a difference.  If I modify the example file to use `PyContext` instead (like in `explore`):
```
    # Create the root context.
    root = PyContext(namespace=person)
    root.environment[Context.TYPE_MANAGER] = type_manager
```
where `person` is an HasTraits object with many nested containers looks like this:
```

person = Person(
    name="foo",
    child=Child(name="bar", mapping={"a": [Book(), Book()]}),
)
```
And if I then applied the fix in https://github.com/enthought/pyface/pull/810 so that it can be launched with wxPython4, I see something like this (after manually expanding the tree nodes):
Screenshot 2020-11-23 at 12 29 55

If we then remove all the adapters in the example, the view looks exactly the same.

**Review by kitchoi — APPROVED** [review]
LGTM.

We will need a news fragment for this change.

**Comment by aaronayres35** [comment]
> * The functionality does work on Python 2 and wxPython 3 (@aaronayres35 please confirm).

Yes, I launched `mayavi2` in this environment (which was a python 2.7 environment) and `explore` was working in the python shell:

```
(mayavi_demo)~/Desktop/mayavi aayres  (master) $ edm list
# Packages in environment 'mayavi_demo'
#  prefix: '/Users/aayres/.edm/envs/mayavi_demo'
appdirs            1.4.3-1     enthought/free
apptools           4.4.0-10    enthought/free
configobj          5.0.6-2     enthought/free
distribute_remove  1.0.0-4     enthought/free
eam                1.1.0-1     enthought/free
eam_mayavi_app     1.0.2-1     enthought/free
enum34             1.1.6-1     enthought/free
envisage           4.7.0-1     enthought/free
libxml2            2.9.4-1     enthought/free
libxslt            1.1.28-7    enthought/free
mayavi             4.5.0-7     enthought/free
mkl                2018.0.3-1  enthought/free
numpy              1.13.3-4    enthought/free
packaging          16.8-2      enthought/free
pyface             6.0.0-1     enthought/free
pygments           2.2.0-1     enthought/free
pyparsing          2.2.0-1     enthought/free
setuptools         38.2.5-1    enthought/free
six                1.10.0-1    enthought/free
traits             4.6.0-1     enthought/free
traitsui           6.0.0-1     enthought/free
vtk                7.0.0-2     enthought/free
wxpython           3.0.2.0-7   enthought/free
```

**Pull request merged**